fastchat/serve/monitor/monitor.py

```diff
@@ -856,6 +857,119 @@ def build_category_leaderboard_tab(
     )
 
 
+def compute_ub_ranking(arena_df):
+    # Sort models based on their scores
+    sorted_models = arena_df.sort_values("score", ascending=False).index.tolist()
+
+    ub_ranking = {}
+    current_rank = 1
+    i = 0
+
+    while i < len(sorted_models):
+        current_model = sorted_models[i]
+        current_lower = arena_df.loc[current_model]["lower"]
+        tied_models = [current_model]
+
+        # Find ties
+        j = i + 1
+        while j < len(sorted_models):
+            next_model = sorted_models[j]
+            if arena_df.loc[next_model]["upper"] >= current_lower:
+                tied_models.append(next_model)
+                j += 1
+            else:
+                break
+
+        # Assign ranks to tied models
+        for model in tied_models:
+            ub_ranking[model] = current_rank
+
+        # Move to the next unprocessed model
+        i = j
+        # Next rank is at least the position in the sorted list
+        current_rank = max(current_rank + 1, i + 1)
+
+    return ub_ranking
+
+
+def process_copilot_arena_leaderboard(leaderboard):
+    leaderboard["score"] = leaderboard["score"].round().astype(int)
+    leaderboard["upper"] = leaderboard["upper"].round().astype(int)
+    leaderboard["lower"] = leaderboard["lower"].round().astype(int)
+
+    leaderboard["upper_diff"] = leaderboard["upper"] - leaderboard["score"]
+    leaderboard["lower_diff"] = leaderboard["score"] - leaderboard["lower"]
+
+    leaderboard["confidence_interval"] = (
+        "+"
+        + leaderboard["upper_diff"].astype(str)
+        + " / -"
+        + leaderboard["lower_diff"].astype(str)
+    )
+
+    rankings_ub = compute_ub_ranking(leaderboard)
+    leaderboard.insert(loc=0, column="Rank* (UB)", value=rankings_ub)
+    leaderboard["Rank"] = leaderboard["score"].rank(ascending=False).astype(int)
+
+    leaderboard = leaderboard.sort_values(by=["Rank"], ascending=[True])
+
+    return leaderboard
+
+
+def build_copilot_arena_tab():
+    copilot_arena_leaderboard_url = "https://leaderboard-server.fly.dev/elo"
```

Comment:
Is there a config file to move this to?

Reply:
There isn't; however, what I've done is move it up to the top, where I see another URL also similarly defined.